zebra-chain/src/orchard/tree.rs

```diff
@@ -317,6 +339,46 @@ impl NoteCommitmentTree {
         }
     }
 
+    /// Returns true if the most recently appended leaf completes the subtree
+    pub fn is_complete_subtree(&self) -> bool {
+        self.inner.value().map_or(false, |x| {
+            x.position()
+                .is_complete_subtree(TRACKED_SUBTREE_HEIGHT.into())
+        })
+    }
+
+    /// Returns subtree index at [`TRACKED_SUBTREE_HEIGHT`]
```

Comment:
Is this the subtree index of the current leaf node, or the most recently completed subtree?

```suggestion
    /// Returns the subtree index for the most recently appended leaf
```